### PR TITLE
275 hastus ID

### LINKED ISSUE #275: Add Hastus ID beside stops to show the ID of those stops that have a hastus ID.
Hastus is 'paitunnus' of stop

### PR DESCRIPTION
**Pull request by kronken**
Closes #275 

Test by: chose any route, for example 0033, and zoom in to a zoom-level of 15 or higher.

screenshot 2018-11-19 at 13 35 54

### CODE REVIEW

File: src/components/map/layers/NodeLayer.tsx [hunk 1]
```diff
@@ -43,31 +49,40 @@ export default class NodeLayer extends Component<MarkerLayerProps> {
             type = node.type;
         }
 
+        if (node.stop && node.stop.hastusId && this.props.mapStore!.zoom >= HASTUS_MIN_ZOOM) {
+            children.push(
+                `<div class=${s.hastusId}>${node.stop.hastusId}</div>`,
+            );
+        }
```
Comment (Divinare) [inline src/components/map/layers/NodeLayer.tsx]: I would take this out from getIcon() method and put it into a getLabel() method instead.

Comment (kronken) [reply src/components/map/layers/NodeLayer.tsx]: That sounds like a good idea

Comment (kronken) [reply src/components/map/layers/NodeLayer.tsx]: fixed

**Comment by Divinare** [comment]
How to test? Imo, a best practice is to add testing instructions to the PR if it is not crystal clear how to test.

For example in here: select a line 550, see hastus ID should be visible on the map of 550's routePath

**Comment by kronken** [comment]
> How to test? Imo, a best practice is to add testing instructions to the PR if it is not crystal clear how to test.
> 
> For example in here: select a line 550, see hastus ID should be visible on the map of 550's routePath

Added

File: src/components/map/layers/NodeLayer.tsx [hunk 2]
```diff
@@ -23,16 +23,28 @@ interface MarkerLayerProps {
...
 const DEFAULT_RADIUS = 25;
+const HASTUS_MIN_ZOOM = 16;
 
 @inject('popupStore', 'toolbarStore', 'sidebarStore', 'mapStore')
 @observer
 export default class NodeLayer extends Component<MarkerLayerProps> {
-    private getMarkerHtml = (markerClass: string) => {
-        return `<div class="${classnames(s.nodeBase, markerClass)}" />`;
+    private getLabelHtml = (text: string) => {
+        return `<div class=${s.hastusId}>
```
Comment (Divinare) [inline src/components/map/layers/NodeLayer.tsx]: rename as hastusLabel

File: src/components/map/layers/NodeLayer.tsx [hunk 3]
```diff
@@ -43,31 +55,40 @@ export default class NodeLayer extends Component<MarkerLayerProps> {
...
-            html = this.getMarkerHtml(s.timeAlignmentMarker);
+            html = this.getMarkerHtml(s.timeAlignmentMarker, children);
             break;
         }
         default: {
-            html = this.getMarkerHtml(isSelected ? s.unknownMarkerHighlight : s.unknownMarker);
+            html = this.getMarkerHtml(
+                isSelected ? s.unknownMarkerHighlight : s.unknownMarker, children);
             break;
         }
```
Comment (Divinare) [inline src/components/map/layers/NodeLayer.tsx]: I think this whole method getIcon() would need rewriting as now it is starting to look complex

according to https://stackoverflow.com/a/34776015 we could aim for a template of:

```
<div>
<div>marker</div>
<div>label</div>
</div>
```

```
getIconHtml() {
    return `
       this.getMarkerHtml()
       this.getLabelHtml()
`
}
```

and have that switch clause in getMarkerHtml & getLabelHtml

**Comment by Divinare** [comment]
Review ^

Testing: ok

**Pull request merged**